Binary files wxWidgets-2.9.2/samples/.DS_Store and wxWidgets-2.9.2-mine/samples/.DS_Store differ
diff -uNr wxWidgets-2.9.2/samples/minimal/minimal.cpp wxWidgets-2.9.2-mine/samples/minimal/minimal.cpp
--- wxWidgets-2.9.2/samples/minimal/minimal.cpp	2011-07-05 01:26:11.000000000 +0400
+++ wxWidgets-2.9.2-mine/samples/minimal/minimal.cpp	2011-08-26 15:42:48.000000000 +0400
@@ -40,6 +40,8 @@
     #include "../sample.xpm"
 #endif
 
+#include <wx/overlay.h>
+
 // ----------------------------------------------------------------------------
 // private classes
 // ----------------------------------------------------------------------------
@@ -60,6 +62,7 @@
 // Define a new frame type: this is going to be our main frame
 class MyFrame : public wxFrame
 {
+    wxOverlay m_overlay;
 public:
     // ctor(s)
     MyFrame(const wxString& title);
@@ -67,6 +70,7 @@
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnIdle(wxIdleEvent& ev);
 
 private:
     // any class wishing to process wxWidgets events must use this macro
@@ -99,6 +103,7 @@
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_IDLE(MyFrame::OnIdle)
 END_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -198,3 +203,13 @@
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnIdle(wxIdleEvent& ev)
+{
+	wxClientDC cdc(this);
+	wxDCOverlay odc(m_overlay, &cdc);
+//	odc.Clear();
+	cdc.SetDeviceOrigin(0, 0);
+	cdc.DrawRectangle(30, 30, 10, 10);
+}
+
